lib/util/git/parent-sha-cache.ts

```diff
@@ -1,15 +1,34 @@
 import { getCache } from '../cache/repository';
+import type { BranchCache } from '../cache/repository/types';
 
 export function getCachedBranchParentShaResult(
-  branchName: string,
-  branchSha: string | null
+  branchName: string
 ): string | null {
   const { branches } = getCache();
   const branch = branches?.find((branch) => branch.branchName === branchName);
 
-  if (branch?.parentSha && branchSha === branch?.sha) {
+  if (branch?.parentSha) {
     return branch.parentSha;
   }
 
   return null;
 }
+
+export function setCachedBranchParentShaResult(
+  branchName: string,
+  parentSha: string
+): void {
+  const cache = getCache();
+  cache.branches ??= [];
+
+  let branch = cache.branches.find(
+    (branch) => branch.branchName === branchName
+  );
+  if (!branch) {
+    branch = {
+      branchName: branchName,
+    } as BranchCache;
```

Comment:
Can this type assert be avoided? It can be risky at some point, i.e. when `BranchCache` is extended with some required field in future.

Reply:
Is there any other way to fix this except from adding all required properties to the new branch object?

Reply:
use `partial<T>()` helper